### PR TITLE
NetBSD support

### PR DESCRIPTION
**Pull request by niacat**

### CODE REVIEW

File: src/SFML/Window/CMakeLists.txt [hunk 1]
```diff
@@ -79,7 +79,8 @@ if(SFML_OS_WINDOWS)
 
     # make sure that we use the Unicode version of the Win API functions
     add_definitions(-DUNICODE -D_UNICODE)
-elseif(SFML_OS_LINUX OR SFML_OS_FREEBSD OR SFML_OS_OPENBSD)
+elseif(SFML_OS_LINUX OR SFML_OS_FREEBSD OR SFML_OS_NETBSD OR SFML_OS_OPENBSD)
+	message("weh")
```
Comment (LaurentGomila) [inline src/SFML/Window/CMakeLists.txt]: Not sure if this is really needed 😄

**Comment by LaurentGomila** [comment]
Is there any significant difference between Open/Free/NetBSD, or could we just merge all that stuff into a generic BSD implemenation?

**Comment by ensra** [comment]
> Is there any significant difference between Open/Free/NetBSD, or could we just merge all that stuff into a generic BSD implementation?

Since the code is filled with ifdef (linux or freebsd or netbsd) you could probably just have a unix system type and handle the special cases within that for different joystick APIs and similar.

**Comment by krytarowski** [comment]
> Is there any significant difference between Open/Free/NetBSD, or could we just merge all that stuff into a generic BSD implementation?

If you want to get a generic switch, go for UNIX or POSIX. BSDs are diverged for over 20 years.

**Comment by eXpl0it3r** [comment]
In the meantime we've gotten another contribution (#1723) with a working Joystick implementation, as such I'll close this PR.

**Pull request closed without merging**